blog-content/monitoring-browser-applications-with-opentelemetry.md

```diff
@@ -0,0 +1,174 @@
+
+If you're interested in a deeper dive on OTeL, check out our recent article: [What is OpenTelemetry and Why Should I Care?](https://highlight.io/blog/what-is-opentelemetry.md).
+
+## OpenTelemetry in Browser Applications
+
+When it comes to browser apps, OpenTelemetry can provide valuable insights into:
+
+1. Document Load Performance
+2. User Interactions
+3. Network Requests (XMLHttpRequest and Fetch)
+
+These insights allow developers to identify bottlenecks, track user interactions, and monitor network requests without manual instrumentation. What's mentioned above is data you get relatively easily by leveraging [OTeL's auto instrumentations](https://opentelemetry.io/docs/zero-code/js/) but you can also add manual instrumentation to collect traces on any other code in you client-side apps.
+
+## Highlight and OpenTelemetry
+
+Highlight's JavaScript SDK offers built-in support for collecting OpenTelemetry data from client-side applications. This integration allows you to seamlessly incorporate OpenTelemetry tracing into your web applications.
+
+### Enabling OpenTelemetry with Highlight
+
+OTeL data collection is still in beta, so you need to explicitly enable it by setting the `enableOtelTracing` configuration option when initializing the SDK:
+
+```ts
+H.init({
+  // ...
+  enableOtelTracing: true
+})
+```
+
+With this simple configuration, Highlight automatically collects most of the OpenTelemetry data you'll need by leveraging the auto instrumentations and doing some additional processing to make the data more useful in Highlight.
+
+### Connecting Client and Server Traces
+
+One of the most powerful features of OpenTelemetry is the ability to connect traces across different services and environments. Highlight's SDKs facilitate this through context propagation, allowing you to create end-to-end traces that span from user interactions in the browser all the way to your backend services.
+
+Here's how it works:
+
+1. **Trace Initiation**: When a user interaction or a network request is initiated in the browser, the SDK creates a new span or continues an existing trace.
+2. **Header Injection**: For outgoing HTTP requests (both XHR and Fetch), the SDK automatically injects trace context headers into the request. These headers typically include:
+   - `traceparent`: Contains the trace ID, parent span ID, and trace flags.
+   - `tracestate`: Carries vendor-specific trace information.
+3. **Server-Side Reception**: The Highlight server-side SDKs extract these headers and continue the trace, linking the server-side spans to the client-side trace.
+4. **Trace Completion**: As the request completes and returns to the client, the full trace, including both client and server spans, can be visualized in Highlight's UI.
+
+This connection between client and server traces provides end-to-end visibility, facilitates performance optimization, and enables error correlation.
```

Comment:
could be good to add the server -> client propagation here as well. that could link to the `Handling Server-Initiated Traces` section below?

For server-side rendering where code executes server side before HTML is sent to the browser, the trace context is propagated through a meta tag that is added to the HTML.